SidecarProfile crashes because of read-only filesystem

### NetBird Operator Version

0.6.0

### Kubernetes Version

1.33.7

### CNI

Cilium

### Describe the bug

I have two clusters set up with the netbird operator. Both can expose Services through NetworkRouters and I can access them.

Now I wanted to access another resource from my Netbird network from within the Kubernetes clusters. For that I have defined a SidecarProfile. I can see that my Pod (that runs a simple Ubuntu image) is getting selected and get's the Netbird-Image as a Sidecar. But the Sidecar is crashing:

```bash
2026-06-10T06:07:12.688Z ERRO client/internal/routemanager/systemops/systemops_linux.go:126: Error setting up sysctl: 1 error occurred:
	* write sysctl net.ipv4.conf.all.src_valid_mark: open /proc/sys/net/ipv4/conf/all/src_valid_mark: read-only file system
2026-06-10T06:07:12.688Z INFO client/internal/routemanager/manager.go:256: Routing setup complete
2026-06-10T06:07:12.734Z INFO client/firewall/create_linux.go:88: creating an nftables firewall manager
2026-06-10T06:07:12.759Z ERRO client/internal/engine.go:546: failed to pull up wgInterface [wt0]: failed to create ipv4 raw socket: socket: operation not permitted
2026-06-10T06:07:12.922Z INFO client/iface/iface.go:336: interface wt0 has been removed
2026-06-10T06:07:13.062Z ERRO client/internal/connect.go:389: error while starting Netbird Connection Engine: up wg interface: failed to create ipv4 raw socket: socket: operation not permitted
```

This happens with the normal Netbird-Image. Switching to the rootless-Image removes the errors, but still doesn't give me access to my other resources from within the Pod. (Access Policy should be fine, I am in the same resource group with my laptop and can access the resource)

I was able to make it work by adding the `NET_RAW` capability to the netbird Sidecar:

```yaml
apiVersion: netbird.io/v1alpha1
kind: SidecarProfile
metadata:
  name: test
spec:
  containerOverride:
    securityContext:
      capabilities:
        add:
          - NET_ADMIN
          - NET_RAW
  setupKeyRef:
    name: sidecar
  podSelector:
    matchLabels:
      app: ubuntu
```

The error message regarding the read-only filesystem still exists though:

```bash
2026-06-10T07:33:55.892Z ERRO client/internal/routemanager/systemops/systemops_linux.go:126: Error setting up sysctl: 1 error occurred:
	* write sysctl net.ipv4.conf.all.src_valid_mark: open /proc/sys/net/ipv4/conf/all/src_valid_mark: read-only file system
```

I also had to switch from the `rootless` to the normal image.

--- a/internal/webhook/v1/pod_webhook.go
+++ b/internal/webhook/v1/pod_webhook.go
@@ -11,6 +11,7 @@ import (
 	"strings"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
@@ -124,6 +125,26 @@ func (d *PodNetbirdInjector) Default(ctx context.Context, pod *corev1.Pod) error
 			Name:  "NB_MANAGEMENT_URL",
 			Value: d.managementURL,
 		},
+		{
+			Name:  "NB_LOG_FILE",
+			Value: "console",
+		},
+		{
+			Name:  "NB_DISABLE_PROFILES",
+			Value: "true",
+		},
+		{
+			Name:  "NB_DISABLE_UPDATE_SETTINGS",
+			Value: "true",
+		},
+		{
+			Name:  "NB_DAEMON_ADDR",
+			Value: "unix:///var/run/netbird/netbird.sock",
+		},
+		{
+			Name:  "NB_ENTRYPOINT_SERVICE_TIMEOUT",
+			Value: "0",
+		},
 	}
 	if len(sidecarProfile.Spec.ExtraDNSLabels) > 0 {
 		envVars = append(envVars, corev1.EnvVar{
@@ -137,8 +158,44 @@ func (d *PodNetbirdInjector) Default(ctx context.Context, pod *corev1.Pod) error
 		Image: d.clientImage,
 		Env:   envVars,
 		SecurityContext: &corev1.SecurityContext{
+			ReadOnlyRootFilesystem: new(true),
 			Capabilities: &corev1.Capabilities{
-				Add: []corev1.Capability{"NET_ADMIN"},
+				Add: []corev1.Capability{
+					"NET_ADMIN",
+					"SYS_RESOURCE",
+					"SYS_ADMIN",
+				},
+			},
+			Privileged: new(true),
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "netbird-run",
+				MountPath: "/var/run/netbird",
+			},
+			{
+				Name:      "netbird-lib",
+				MountPath: "/var/lib/netbird",
+			},
+			{
+				Name:      "ssh-etc",
+				MountPath: "/etc/ssh",
+			},
+			{
+				Name:      "resolv-conf",
+				MountPath: "/etc/resolv.conf",
+				SubPath:   "resolv.conf",
+			},
+			{
+				Name:      "resolv-conf",
+				MountPath: "/etc/resolv.conf.original.netbird",
+				SubPath:   "resolv.conf.original.netbird",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("100m"),
+				corev1.ResourceMemory: resource.MustParse("128Mi"),
 			},
 		},
 	}
@@ -160,6 +217,53 @@ func (d *PodNetbirdInjector) Default(ctx context.Context, pod *corev1.Pod) error
 			return err
 		}
 	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: "netbird-run",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: "netbird-lib",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: "ssh-etc",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: "resolv-conf",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, volumes...)
+
+	resolvInitContainer := corev1.Container{
+		Name:    "resolv-conf",
+		Image:   d.clientImage,
+		Command: []string{"sh", "-c", "cp /etc/resolv.conf /tmp/resolv.conf && cp /etc/resolv.conf /tmp/resolv.conf.original.netbird"},
+		SecurityContext: &corev1.SecurityContext{
+			ReadOnlyRootFilesystem: new(true),
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{"ALL"},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "resolv-conf",
+				MountPath: "/tmp",
+			},
+		},
+	}
+	pod.Spec.InitContainers = append(pod.Spec.InitContainers, resolvInitContainer)
 
 	switch sidecarProfile.Spec.InjectionMode {
 	case nbv1alpha1.InjectionModeSidecar:

